src/en/coloredmanga/src/eu/kanade/tachiyomi/extension/en/coloredmanga/ColoredManga.kt

```diff
@@ -0,0 +1,395 @@
+                            statusExcluded.add(tri.value)
+                        }
+                    }
+                    else -> {}
+                }
+            }
+
+            val includeGenre = genreIncluded.isEmpty() || it.tags.map { genre -> genre.lowercase() }.containsAll(genreIncluded)
+            val excludeGenre = genreExcluded.isNotEmpty() && it.tags.map { genre -> genre.lowercase() }.containsAll(genreExcluded)
+
+            val includeType = typeIncluded.isEmpty() || typeIncluded.contains(it.type.lowercase())
+            val excludeType = typeExcluded.isNotEmpty() && typeExcluded.contains(it.type)
+
+            val includeColor = colorIncluded.isEmpty() || colorIncluded.contains(it.version)
+            val excludeColor = colorExcluded.isNotEmpty() && colorExcluded.contains(it.version)
+
+            val regularSearch = it.name.contains(title) || it.synopsis.contains(title)
+            includeGenre && !excludeGenre &&
+                includeType && !excludeType &&
+                includeColor && !excludeColor &&
+                regularSearch
+        }
+
+        val sorted = when (sort.first) {
+            "pop" -> {
+                if (sort.second == "desc") {
+                    mangas.sortedByDescending { it.totalViews }
+                } else {
+                    mangas.sortedBy { it.totalViews }
+                }
+            }
+            "tit" -> {
+                if (sort.second == "desc") {
+                    mangas.sortedByDescending { it.name }
+                } else {
+                    mangas.sortedBy { it.name }
+                }
+            }
+            else -> {
+                if (sort.second == "desc") {
+                    mangas.sortedByDescending {
+                        try {
+                            dateFormat.parse(it.date)!!.time
+                        } catch (e: Exception) {
+                            0L
+                        }
+                    }
+                } else {
+                    mangas.sortedBy {
+                        try {
+                            dateFormat.parse(it.date)!!.time
+                        } catch (e: Exception) {
+                            0L
+                        }
+                    }
+                }
+            }
+        }
+
+        val final = sorted.map(::popularManga)
+        return MangasPage(final, false)
+    }
+
+    private fun getMangas(doc: Document): MutableList<Mangas> {
+        val mangasRaw = doc.selectFirst("script:containsData(\\\"cover)")!!.data()
+        val mangasJson = mangasRaw.replace("\\\"", "\"").replace("\\\\\"", "\\\"").replace("self.__next_f.push([1,\"7:[\"\$\",\"div\",null,{\"className\":\"xl:w-[1280px] w-full min-h-[100vh] font-light flex flex-col items-start justify-start\",\"id\":\"themes_body___1nq3\",\"children\":[\"\$\",\"\$Le\",null,{\"params\":\"\$undefined\",\"data\":[", "{\"data\":[").replace("}]}]}]\\n\"])", "}]}")

```

Comment:
this can break very easily, either use regex or substrings. also maybe they have an api which can be scraped

Reply:
They do have api, I think. But I couldn't understand how it works as the source js uses alphabets and nunbers as variables/function name.

Reply:
maybe [this](https://github.com/manga-download/haruneko/pull/657) will help

Reply:
Thank you, but can I just use replaceBefore to get the data json?:
`val mangasJson = mangasRaw
            .replace(Regex("""\\([\\"])"""), "$1")
            .replaceBefore("\"data\":[", "{")
            .removeSuffix("]}]\\n\"])")`